src/dmqnode/storage/engine/DiskOverflow.d

```diff
@@ -890,7 +890,7 @@ class DiskOverflow: DiskOverflowInfo
         log.info("Renaming channel {} '{}' -> '{}'", channel.id, old_name, new_name);
 
         this.channels[new_name] = *channel;
-        this.channels.remove(old_name);
+        this.channels.remove(cast(istring)old_name);
```

Comment:
I understand the reluctance against the cast, but I think non-immutable AA keys are even worse. Consequently the code would end up in a situation like this:
```d
ChannelMetadata[cstring] channels;

void addChannel ( cstring channel_name )
{
    channels[channel_name] = channel;
}
```
Compared to the `cast(istring)` upon `remove`, which looks ugly but is in fact safe since `remove` is well known to not store the argument, this is _really_ unsafe.

Reply:
>  Consequently the code would end up in a situation like this

Built-in AA API is pretty much broken when it comes to const. This works despite your expectations:

```D
void main ( )
{
    int[string] aa;
    const(char)[] text = "abcd";
    aa[text] = 42;
}
```

Because of that, it only makes sense to declare AA keys as `const`, not `immutable` - you won't be able to restrict assignment of new keys anyway, but this way at least lookup/foreach type deduction will work for you.

Reply:
That is not my point. My point is to prevent using mutable key strings because they can be modified after assigning them to the AA. This has happened before in the dmqnode. With `cstring` it is possible while `istring` prevents it.

Reply:
Yeah, and I am trying to show that it doesn't prevent it:

```D
void main ( )
{
    int[string] aa;
    char[] text = "abcd".dup;
    const(char)[] text_c = text;
    aa[text_c] = 42;
    text[0] = 'x';
}
```

Reply:
(yes, it is a huge bug in compiler/runtime; no, it is unlikely to get fixed until full rework of how AA are implemented)

Reply:
So relying on a bug is better than a `cast`? :P

Reply:
Sure. One simply can't use AAs without casts otherwise, so as long as API is present as it is, there is not other option.